x/dsmr/block.go

```diff
@@ -79,41 +75,46 @@ func signChunk[T Tx](
 	copy(pkBytes[:], bls.PublicKeyToCompressedBytes(pk))
 	copy(signature[:], signatureBytes)
 
-	return newChunk(chunk, pkBytes, signature)
+	return newChunk(chunk, pkBytes, signature), nil
 }
 
 // newChunk signs a chunk
-func newChunk[T Tx](
+func newChunk[T Tx[T]](
 	unsignedChunk UnsignedChunk[T],
 	signer [bls.PublicKeyLen]byte,
 	signature [bls.SignatureLen]byte,
-) (Chunk[T], error) {
+) Chunk[T] {
 	c := Chunk[T]{
 		UnsignedChunk: unsignedChunk,
 		Signer:        signer,
 		Signature:     signature,
 	}
-	return c, c.init()
+	c.init()
+	return c
 }
 
-func ParseChunk[T Tx](chunkBytes []byte) (Chunk[T], error) {
-	c := Chunk[T]{}
-	if err := codec.LinearCodec.UnmarshalFrom(&wrappers.Packer{Bytes: chunkBytes}, &c); err != nil {
+func ParseChunk[T Tx[T]](chunkBytes []byte) (Chunk[T], error) {
+	var c Chunk[T]
+	if err := c.UnmarshalCanoto(chunkBytes); err != nil {
 		return Chunk[T]{}, err
 	}
-
-	return c, c.init()
+	c.init()
+	return c, nil
 }
 
 type BlockHeader struct {
-	ParentID  ids.ID `serialize:"true"`
-	Height    uint64 `serialize:"true"`
-	Timestamp int64  `serialize:"true"`
+	ParentID  ids.ID `canoto:"fixed bytes,1"`
+	Height    uint64 `canoto:"int,2"`
+	Timestamp int64  `canoto:"int,3"`
+
+	canotoData canotoData_BlockHeader
 }
 
 type Block struct {
 	BlockHeader
```

Comment:
Should this `BlockHeader` be serialized?

Reply:
I think this will technically pass tests but we should serialize here because the block id/bytes should be derived from the `BlockHeader`